src/Acuminator/Acuminator.Utilities/Common/EnumerableExtensions.cs

```diff
@@ -138,6 +138,33 @@ private static IEnumerable<T> ConcatIterator<T>(T extraElement, IEnumerable<T> s
 			}
 		}
 
+		/// <summary>
+		/// Where method for <see cref="SyntaxTokenList"/>. This is an optimization method which allows to avoid boxing.
+		/// </summary>
+		/// <param name="source">The <see cref="SyntaxTokenList"/> to act on.</param>
+		/// <param name="predicate">The predicate.</param>
+		/// <returns/>
+		[DebuggerStepThrough]
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static IEnumerable<SyntaxToken> Where(this SyntaxTokenList source, Func<SyntaxToken, bool> predicate)
+		{
+			predicate.ThrowOnNull(nameof(predicate));
+			return WhereForSyntaxTokenListImplementation();
+
+			IEnumerable<SyntaxToken> WhereForSyntaxTokenListImplementation()
```

Comment:
What is the need to create a local function here?

Reply:
It is a common pattern which provides immediate arguments check by moving iterator into a separate method. 
If I did everything in one method and someone passed `null` for predicate then the ArgumentNullException will be called only when someone tried to iterate the results which can happen in a place quite distant from the method call

Reply:
Got it, thanks.
Does the `[MethodImpl(MethodImplOptions.AggressiveInlining)]` work in this case?

Reply:
It's actually difficult to say - it really depends because method inlining is JIT heuristic
There are some constraints on methods that could be inlined:
https://dzone.com/articles/aggressive-inlining-clr-45-jit
This flag removes the constraint on method size (no more than 32 bytes), so it seemed to worth to try it. Bu it doesn't guarantee inline. Moreover, the context of the method call also matters:
https://www.codeproject.com/Tips/1072041/NET-Methods-Inlining-and-Loops